src/riak_governor_ensemble_master.erl

```diff
@@ -75,7 +89,12 @@ ensure_ensembles_started(#state{ensemble_size=EnsembleSize}) ->
     Ensembles0 = all_ensembles(EnsembleSize),
     ClusterNodes = riak_governor_util:get_cluster_nodes(),
     Ensembles = [ClusterNodes|Ensembles0],
-    [ensure_ensemble_started(Nodes) || Nodes <- Ensembles].
+    Res = [ensure_ensemble_started(Nodes) || Nodes <- Ensembles],
+    % validate all results are ok
+    case lists:usort(Res) of
+        [ok] -> ok;
+        _ -> {error, {unexpected_ensemble_start_results, Res}}
+    end.
```

Comment:
But, how riak_ensemble behaves here? If it also returns `already[_started | _exists]` then you may have infinite loop.